From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Raine Makelainen <raine.makelainen@jolla.com>
Date: Wed, 19 Aug 2020 14:27:21 +0300
Subject: [PATCH 52/67] Enable fullscreen for EmbedLite. Fixes JB#44129

EmbedLite's root docshell is not chrome.

Keep the upstream rejection for non-Qt toolkits.

Let Qt embedding continue through the fullscreen path.

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 dom/base/nsGlobalWindowOuter.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dom/base/nsGlobalWindowOuter.cpp b/dom/base/nsGlobalWindowOuter.cpp
index e916356d774d6..07cd4eef443e0 100644
--- a/dom/base/nsGlobalWindowOuter.cpp
+++ b/dom/base/nsGlobalWindowOuter.cpp
@@ -4223,10 +4223,14 @@ nsresult nsGlobalWindowOuter::SetFullscreenInternal(FullscreenReason aReason,
   if (rootItem != mDocShell)
     return window->SetFullscreenInternal(aReason, aFullscreen);
 
+  // EmbedLite's root docshell is not chrome. Allow Qt embedding to continue
+  // through the fullscreen path without changing the docshell item type.
+#ifndef MOZ_WIDGET_QT
   // make sure we don't try to set full screen on a non-chrome window,
   // which might happen in embedding world
   if (mDocShell->ItemType() != nsIDocShellTreeItem::typeChrome)
     return NS_ERROR_FAILURE;
+#endif
 
   // FullscreenReason::ForForceExitFullscreen can only be used with exiting
   // fullscreen
